tests/unit/test_trial_component.py

```diff
@@ -226,6 +226,21 @@ def test_delete(sagemaker_boto_client):
     sagemaker_boto_client.delete_trial_component.assert_called_with(TrialComponentName="foo")
 
 
+def test_delete_with_disassociate(sagemaker_boto_client):
+    obj = trial_component.TrialComponent(sagemaker_boto_client, trial_component_name="foo", display_name="bar")
+    sagemaker_boto_client.delete_trial_component.return_value = {}
+    sagemaker_boto_client.list_trials.return_value = {
+        "TrialSummaries": [{"TrialName": "trial-1"}, {"TrialName": "trial-2"},]
+    }
+    obj.delete(disassociate=True)
```

Comment:
add a test that calls with False or add an assertion in an existing test that verifies disassociate is not called

Reply:
test_create_delete() should cover it?